app/lib/events/event.rb

```diff
@@ -0,0 +1,32 @@
+module Events
+  class Event
+    def initialize
+      @event_hash = {
+        environment: HostingEnvironment.environment_name,
+      }
+    end
+
+    def as_json
+      @event_hash.as_json
+    end
+
+    def with_request_details(rack_request)
+      @event_hash.merge!(
+        request_uuid: rack_request.uuid,
+        request_path: rack_request.path,
+        request_method: rack_request.method,
+      )
+
+      self
+    end
+
+    def with_user_and_namespace(user, namespace)
+      @event_hash.merge!(
+        namespace: namespace,
+        user_id: user&.id,
+      )
```

Comment:
Do we have `support_ref` at this point? That's a key identifier used in the existing reports, could be useful here.

Reply:
only for candidates — what's here is the lowest common denominator between namespaces. Would it makes sense to get the machinery in & ship this structure, then have another conversation about the schema?

Reply:
Ya, I think one step at a time works.